qcodes/utils/validators.py

```diff
@@ -58,51 +67,54 @@ class Validator:
 
         raises an error (TypeError or ValueError) if the value fails
 
+    is_numeric: A boolean flag that marks if this a numeric type.
+
+    The base class implements:
+
     valid_values: a property exposing _valid_values, which is a list
         of examples of valid values. For very simple validators, like
         Bool or Enum, the list contains all valid values, but in general
         it just holds SOME valid values. These example values are intended
         to be useful when simulating instruments.
 
-    is_numeric: is this a numeric type (so it can be swept)?
+    Alternatively you may override valid_values and provide your own
+    implementation of getting valid values.
     """
+    _valid_values: Any = None
+    is_numeric = False  # is this a numeric type (so it can be swept)?
 
-    def __init__(self):
-        raise NotImplementedError
-
-    def validate(self, value, context=''):
+    def validate(self, value, context: str=''):
         raise NotImplementedError
 
     @property
-    def valid_values(self):
-        if not hasattr(self, '_valid_values'):
-            raise NotImplementedError
-        else:
+    def valid_values(self) -> TList[Any]:
+        if hasattr(self, '_valid_values'):
             return self._valid_values
```

Comment:
what is the point of this clause if `_valid_values` exists in the "base" class?

Reply:
I thin you are right. This was needed before the last refactor but is no longer needed.